### PR TITLE
Blog post for running bulk jobs in OpenStack

### PR DESCRIPTION
**Pull request by cloudnull**
This post covers running bulk jobs and commits across many
repositories governed under a single OpenStack project.

Signed-off-by: Kevin Carter <kevin.carter@rackspace.com>

In a comment on the pull request, include the following information and check off the list.

 - [x] Mini bio
  - Soft capped to 75 words
 - [x] Desired date of publishing
  - Make sure this matches the `date` in the front matter
 - [x] Social media contact info
 - [x] Redaction: (yes|no)
  - Allow redaction, editing or structure changes by technical writer staff
 - [x] Comments: (yes|no)
  - If you're willing to answer comments about your post, set `comments: true`.
  - Make sure to subscribe to your post comments by clicking on the **Subscribe** link at the very bottom of the post _once_ it's published.

@catlook Please review.

### CODE REVIEW

**Comment by cloudnull** [comment]
Mini bio:
 
  * Kevin Carter is a Rackspace Private Cloud developer, Cloud Builder, Loves Open Source, and Hates Nonsense, works on OpenStack,  (I know, those last three are closely related). 

Desired date of publishing: 2017-03-30 (whenever)

Social media contact info:

   * twitter: @cloudnull

Redaction: **Yes**.

Comments: Makes no difference to me, presently ``false``.

**Comment by asettle** [comment]
@catlook - any reason why we can't see the failed tests? It appears to be telling us we're unauthorised.

**Comment by catlook** [comment]
Did you log into strider @asettle ?

**Comment by catlook** [comment]
Looks like the categories you used won't work. If you remove those categories, it should build.

Category 'puppet' only has 7 posts
Category 'ruby' only has 7 posts
Category 'public-cloud' only has 7 posts
Category 'rpc' only has 7 posts
Category 'openstack-ansible' only has 1 post
Category 'rpc-o' only has 1 post

**Comment by cloudnull** [comment]
@catlook is there a set of categories I should use or should I remove that section entirely?

Even with just the one category, which my colleague used a little while back, it seem to be unhappy 
``` diff
+categories:
 +    - OpenStack
```

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+categories:
+    - OpenStack
+---
+
+As OpenStack projects grow its likely that a given project will spawn
+several repositories and will share code snippets across them.
+
+<!-- more -->
+
+If you're ever in need of updating a mess of repositories all at the
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... time, these steps, and the subsequent ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+same time these couple of steps, and subsequent script, can help you
+identify all repositories governed under a single project and run some
+type of bulk job(s) across them.
+
+----
+
+##### Getting a list of all repositories under a given project's governance.
+
+First clone OpenStack Infra's **project-config** repository.
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: First, clone ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+``` bash
+git clone https://github.com/openstack-infra/project-config project-config
+```
+
+Change into the ``project-config`` directory and then use the following bit
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Next, change ...
... directory, and then use ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+for project in projects:
+    pgs = project.get('groups', list())
+    dig = project.get('docimpact-group', 'unknown')
+    if PROJECT_GROUP in pgs or PROJECT_GROUP in dig:
+        project_entry = project['project']
+        project_github = 'https://github.com/%s' % project['project']
+        print(project_github)
+```
+
+The previous snippet will print the github URL for all the repositories. If
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... snippet prints the ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+this was fed into a bash script you could simply loop over the results taking
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... script, you ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+the standard out and to clone the repository.
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... standard out and cloning each repository.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    - OpenStack
+---
+
+As OpenStack projects grow its likely that a given project will spawn
+several repositories and will share code snippets across them.
+
+<!-- more -->
+
+If you're ever in need of updating a mess of repositories all at the
+same time these couple of steps, and subsequent script, can help you
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: same time, these steps, and the linked script, can ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 2 above)
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: First, clone ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 3 above)
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Next, change ...
... directory, and then use ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+Once cloned, enter into the cloned directory and run whatever bulk job is
+needed.
+
+----
+
+
+##### Running bulk jobs using a script.
+
+Here's a [simple script](https://github.com/cloudnull/random_scripts/blob/master/openstacky-things/bulk-job-doer.sh)
+which pulls all of this together and can serve as a basic template when
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... template for

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+running bulk jobs across a sprawl of repositories. The following script has
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: ... The linked script ...

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
+---
+layout: post
+title: OpenStack Bulk Job Runner
+date: 2017-03-27 22:22
```
Comment (catlook) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: [Please change this to the date on which you want the blog published!]

**Comment by catlook** [comment]
Kevin, let me see if there's another back-end problem with the categories.  I do know it's based on the percentage of papers using the category.  So if more papers have been published, the numbers using a specific category like "Openstack" may fall below the threshold.  I'll see what we can do!

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 7 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 10 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 2 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 3 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 5 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 6 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: Done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 9 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 8 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: done.

File: _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md (hunk 4 above)
Comment (cloudnull) [inline _posts/2017-03-27-OpenStack-Bulk-Job-Runner.md]: done.

**Comment by catlook** [comment]
Thanks, K!  Troubleshooting the build and will publish as soon as I can!

**Comment by catlook** [comment]
We are published!! Thanks for your patience!!
https://developer.rackspace.com/blog/OpenStack-Bulk-Job-Runner/

**Pull request merged**